### ISSUE TITLE
Suggested change to compile Kokkos 3.7 with musl-libc

### ISSUE THREAD

**Issue by akohlmey** [kokkos/kokkos#5869]
**Describe the bug**
To compile Kokkos 3.7 on a system running musl-libc instead of GNU libc the following change can be applied.
```
  diff --git a/lib/kokkos/core/src/Kokkos_Macros.hpp b/lib/kokkos/core/src/Kokkos_Macros.hpp
  index 8c2d414a99..3370d908ea 100644
  --- a/lib/kokkos/core/src/Kokkos_Macros.hpp
  +++ b/lib/kokkos/core/src/Kokkos_Macros.hpp
  @@ -650,7 +650,9 @@ static constexpr bool kokkos_omp_on_host() { return false; }
   #if (defined(KOKKOS_COMPILER_GNU) || defined(KOKKOS_COMPILER_CLANG) ||  \
        defined(KOKKOS_COMPILER_INTEL) || defined(KOKKOS_COMPILER_PGI)) && \
       !defined(_WIN32)
  +#if (defined(__linux__) && defined(__GLIBC_MINOR__))
   #define KOKKOS_IMPL_ENABLE_STACKTRACE
  +#endif
   #define KOKKOS_IMPL_ENABLE_CXXABI
   #endif
   
```

There is an equivalent change in upstream that uses a C++17 only feature, so we have implemented the above change in the copy bundled with LAMMPS so we can stick with the C++14 requirement. We use the same heuristic and a few other places in LAMMPS with success and rely the fact that musl-libc is Linux-only and is only partially trying to appear like GNU libc.

This is a suggestion to add to the 3.7 maintenance branch since develop is already fixed and needs C++17 anyway.

**Comment by dalg24** [comment]
For the record, Axel is referring to #5678 which is in the current RC 4.0

**Comment by dalg24** [comment]
> musl-libc is Linux-only and is only partially trying to appear like GNU libc

Should the guard not read `#if !defined(__linux__) || defined(__GLIBC_MINOR__)` then?

**Comment by akohlmey** [comment]
> > musl-libc is Linux-only and is only partially trying to appear like GNU libc
> 
> Should the guard not read `#if !defined(__linux__) || defined(__GLIBC_MINOR__)` then?

Good point. I don't usually get to see stacktraces and for compiling on the machines I am compiling with Kokkos on it doesn't make a difference. I will make the corresponding change in LAMMPS.

**Comment by dalg24** [comment]
The fix will be in the 3.7.2 patch release.
